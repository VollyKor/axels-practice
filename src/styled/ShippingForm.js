import styled from "styled-components";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/button";

import { BiTargetLock } from "react-icons/bi";

export const FormPhoneDesc = styled.p`
  font-size: 14px;
  line-height: 1;
`;

export const FindButton = styled.button`
  position: absolute;
  z-index: 1;

  right: 0;
  top: 4px;

  background-color: transparent;
  border: none;
`;

export const CityInput = () => (
  <div className="position-relative">
    <Form.Group controlId="city" className="position-relative">
      <Form.Control type="text" placeholder="City" />
    </Form.Group>
    <FindButton type="button">
      <BiTargetLock size="25px" color="#5a1094" />
    </FindButton>
  </div>
);

export const SubmitButton = styled(Button)`
  outline: none;
  border: none;
  background-color: var(--primary);
`;
