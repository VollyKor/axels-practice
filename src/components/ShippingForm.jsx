import { Form, Col } from "react-bootstrap";

import { FormPhoneDesc, CityInput, SubmitButton } from "../styled/ShippingForm";

export default function ShippingForm() {
  return (
    <>
      <h2 className="ml-4">Shipping Info</h2>
      <Form className="p-4">
        <Form.Label>Recipient</Form.Label>
        <Form.Group controlId="name">
          <Form.Control type="name" placeholder="Full name" />
        </Form.Group>
        <Form.Row className="align-items-center">
          <Col lg={6}>
            <Form.Group controlId="phone">
              <Form.Control type="phone" placeholder="Daytime Phone" />
            </Form.Group>
          </Col>

          <Col lg={4}>
            <FormPhoneDesc className="text-muted">
              For delivery questions only
            </FormPhoneDesc>
          </Col>
        </Form.Row>

        <Form.Label>Address</Form.Label>

        <Form.Group controlId="address">
          <Form.Control type="text" placeholder="Street Address" />
        </Form.Group>

        <Form.Group controlId="gate-code">
          <Form.Control
            type="text"
            placeholder="Apt, Suit, Bldg, Gate Code. (optional)"
          />
        </Form.Group>

        <CityInput />

        <Form.Row>
          <Col lg={7}>
            <Form.Group controlId="country">
              <Form.Control type="text" placeholder="Country" />
            </Form.Group>
          </Col>

          <Col lg={5}>
            <Form.Group controlId="zip">
              <Form.Control type="text" placeholder="ZIP" />
            </Form.Group>
          </Col>
        </Form.Row>
        <SubmitButton type="submit" className="px-4 px-sm-5 py-2">
          Continue
        </SubmitButton>
      </Form>
    </>
  );
}
