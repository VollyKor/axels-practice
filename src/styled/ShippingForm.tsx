import styled from 'styled-components';
import Button from 'react-bootstrap/button';

export const FormPhoneDesc = styled.p`
    font-size: 14px;
    line-height: 1;
`;

export const SubmitButton = styled(Button)`
    outline: none;
    border: none;
    background-color: var(--primary);
`;
