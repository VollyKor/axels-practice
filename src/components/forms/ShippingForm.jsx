import { useRef, useEffect, useState, useMemo } from 'react';
import { Form, Col } from 'react-bootstrap';
import { useFormik } from 'formik';
import { Tooltip } from 'components';
import { usePlacesWidget } from 'react-google-autocomplete';
import countryList from 'react-select-country-list';

import { FormPhoneDesc, CityInput, SubmitButton } from 'styled/ShippingForm';

import {
    formType,
    initialShippingFormValues,
    fieldName,
    localStorageKey,
} from 'helpers/constants';
import getGeo from 'helpers/getGeo';
import { shippingFormValidate } from 'helpers/validationSchemas';

const { REACT_APP_GOOGLE_API_KEY } = process.env;

export default function ShippingForm({ setCurrentForm }) {
    const [geoposition, setGeoposition] = useState(null);
    const [address, setAddress] = useState(null);

    const options = useMemo(() => countryList().getData(), []);
    console.log(options);
    const fullNameRef = useRef(null);
    const phoneRef = useRef(null);
    const addressRef = useRef(null);
    const cityRef = useRef(null);
    const GateCodeRef = useRef(null);
    const countryRef = useRef(null);
    const zipCodeRef = useRef(null);

    const {
        handleSubmit,
        setFieldValue,
        getFieldProps,
        touched,
        errors,
        resetForm,
    } = useFormik({
        initialValues: initialShippingFormValues,
        onSubmit: (FormData) => {
            localStorage.setItem(
                localStorageKey.shippingForm,
                JSON.stringify(FormData)
            );
            setCurrentForm(formType.Billing);
        },
        validationSchema: shippingFormValidate,
    });

    const { ref } = usePlacesWidget({
        apiKey: REACT_APP_GOOGLE_API_KEY,
        options: {
            types: ['(country)'],
            componentRestrictions: { country: 'ua' },
        },
        onPlaceSelected: (place) => {
            console.log(place);
            setFieldValue(fieldName.country, place.formatted_address);
        },
    });

    useEffect(() => {
        if (!geoposition) {
            navigator.geolocation.getCurrentPosition((position) =>
                setGeoposition(position)
            );
        }

        getGeo(geoposition, setAddress);
    }, [geoposition]);

    useEffect(() => {
        if (address) {
            resetForm({ values: { ...initialShippingFormValues, ...address } });
        }
    }, [address, resetForm]);

    const getWarningStyleBg = (field) =>
        errors[field] && touched[field] && 'bg-warning';

    return (
        <>
            <h2 className="ml-4">Shipping Info</h2>
            <Form className="p-4" onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Control ref={ref} />
                </Form.Group>
                <Form.Label>Recipient</Form.Label>

                <Form.Group>
                    <Form.Control
                        ref={fullNameRef}
                        type="name"
                        placeholder="Full name"
                        {...getFieldProps(fieldName.fullName)}
                        className={getWarningStyleBg(fieldName.fullName)}
                    />
                    <Tooltip
                        fieldName={fieldName.fullName}
                        forwardRef={fullNameRef}
                        touched={touched}
                        errors={errors}
                    />
                </Form.Group>

                <Form.Row className="align-items-center">
                    <Col lg={6}>
                        <Form.Group>
                            <Form.Control
                                type="phone"
                                ref={phoneRef}
                                placeholder="Daytime Phone"
                                className={getWarningStyleBg(fieldName.phone)}
                                {...getFieldProps(fieldName.phone)}
                            />

                            <Tooltip
                                fieldName={fieldName.phone}
                                forwardRef={phoneRef}
                                touched={touched}
                                errors={errors}
                            />
                        </Form.Group>
                    </Col>

                    <Col lg={4}>
                        <FormPhoneDesc className="text-muted">
                            For delivery questions only
                        </FormPhoneDesc>
                    </Col>
                </Form.Row>

                <Form.Label>Address</Form.Label>

                <Form.Group>
                    <Form.Control
                        ref={addressRef}
                        type="text"
                        placeholder="Street Address"
                        {...getFieldProps(fieldName.address)}
                        className={getWarningStyleBg(fieldName.address)}
                    />

                    <Tooltip
                        fieldName={fieldName.address}
                        forwardRef={addressRef}
                        touched={touched}
                        errors={errors}
                    />
                </Form.Group>

                <Form.Group>
                    <Form.Control
                        type="text"
                        ref={GateCodeRef}
                        placeholder="Apt, Suit, Bldg, Gate Code. (optional)"
                        {...getFieldProps(fieldName.gateCode)}
                        className={getWarningStyleBg(fieldName.gateCode)}
                    />

                    <Tooltip
                        fieldName={fieldName.gateCode}
                        forwardRef={GateCodeRef}
                        touched={touched}
                        errors={errors}
                    />
                </Form.Group>

                <CityInput
                    inputRef={cityRef}
                    className={getWarningStyleBg(fieldName.city)}
                    {...getFieldProps(fieldName.city)}
                />
                <Tooltip
                    fieldName={fieldName.city}
                    forwardRef={cityRef}
                    touched={touched}
                    errors={errors}
                />

                <Form.Row>
                    <Col lg={7}>
                        <Form.Group>
                            <Form.Control
                                as="select"
                                type="text"
                                ref={countryRef}
                                placeholder="Country"
                                className={getWarningStyleBg(fieldName.country)}
                                {...getFieldProps(fieldName.country)}
                            >
                                {options.map((el) => (
                                    <option key={el.label}>{el.label}</option>
                                ))}
                            </Form.Control>

                            <Tooltip
                                fieldName={fieldName.country}
                                forwardRef={countryRef}
                                touched={touched}
                                errors={errors}
                            />
                        </Form.Group>
                    </Col>

                    <Col lg={5}>
                        <Form.Group controlId="zip">
                            <Form.Control
                                type="text"
                                ref={zipCodeRef}
                                placeholder="ZIP"
                                className={getWarningStyleBg(fieldName.zip)}
                                {...getFieldProps(fieldName.zip)}
                            />

                            <Tooltip
                                fieldName={fieldName.zip}
                                forwardRef={zipCodeRef}
                                touched={touched}
                                errors={errors}
                            />
                        </Form.Group>
                    </Col>
                </Form.Row>

                <SubmitButton type="submit" className="px-4 px-sm-5 py-2">
                    Continue
                </SubmitButton>
            </Form>
        </>
    );
}
