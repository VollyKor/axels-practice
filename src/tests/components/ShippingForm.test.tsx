import { shallow } from 'enzyme';

import ShippingForm from 'components/forms/ShippingForm';

it('renders snapshot', () => {
    const wrapper = shallow(<ShippingForm />);
    expect(wrapper).toMatchSnapshot();
});
