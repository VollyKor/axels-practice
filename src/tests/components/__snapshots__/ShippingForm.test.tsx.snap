// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<Fragment>
  <h2
    className="ml-4 h4"
  >
    Shipping Info
  </h2>
  <Form
    className="px-1 px-md-4 pt-1 pb-5"
    inline={false}
    onSubmit={[Function]}
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Recipient
    </FormLabel>
    <FormGroup>
      <FormControl
        name="fullName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Full name"
        type="name"
        value=""
      />
      <Tooltip
        errors={Object {}}
        fieldName="fullName"
        forwardRef={
          Object {
            "current": null,
          }
        }
        touched={Object {}}
      />
    </FormGroup>
    <FormRow
      className="align-items-center"
    >
      <Col
        lg={7}
      >
        <FormGroup>
          <InputElement
            mask="(999) 999-99-99"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <Component />
          </InputElement>
          <Tooltip
            errors={Object {}}
            fieldName="phone"
            forwardRef={
              Object {
                "current": null,
              }
            }
            touched={Object {}}
          />
        </FormGroup>
      </Col>
      <Col
        lg={4}
      >
        <styled.p
          className="text-muted"
        >
          For delivery questions only
        </styled.p>
      </Col>
    </FormRow>
    <FormLabel
      column={false}
      srOnly={false}
    >
      Address
    </FormLabel>
    <FormGroup>
      <FormControl
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Street Address"
        type="text"
        value=""
      />
      <Tooltip
        errors={Object {}}
        fieldName="address"
        forwardRef={
          Object {
            "current": null,
          }
        }
        touched={Object {}}
      />
    </FormGroup>
    <FormGroup>
      <FormControl
        name="gateCode"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Apt, Suit, Bldg, Gate Code. (optional)"
        type="text"
        value=""
      />
      <Tooltip
        errors={Object {}}
        fieldName="gateCode"
        forwardRef={
          Object {
            "current": null,
          }
        }
        touched={Object {}}
      />
    </FormGroup>
    <CityInput
      fillForm={[Function]}
      inputRef={
        Object {
          "current": null,
        }
      }
      name="city"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    />
    <Tooltip
      errors={Object {}}
      fieldName="city"
      forwardRef={
        Object {
          "current": null,
        }
      }
      touched={Object {}}
    />
    <FormRow>
      <Col
        lg={7}
      >
        <FormGroup>
          <FormControl
            as="select"
            name="country"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Country"
            type="text"
            value=""
          >
            <option />
            <option
              key="Afghanistan"
            >
              Afghanistan
            </option>
            <option
              key="Åland Islands"
            >
              Åland Islands
            </option>
            <option
              key="Albania"
            >
              Albania
            </option>
            <option
              key="Algeria"
            >
              Algeria
            </option>
            <option
              key="American Samoa"
            >
              American Samoa
            </option>
            <option
              key="Andorra"
            >
              Andorra
            </option>
            <option
              key="Angola"
            >
              Angola
            </option>
            <option
              key="Anguilla"
            >
              Anguilla
            </option>
            <option
              key="Antarctica"
            >
              Antarctica
            </option>
            <option
              key="Antigua and Barbuda"
            >
              Antigua and Barbuda
            </option>
            <option
              key="Argentina"
            >
              Argentina
            </option>
            <option
              key="Armenia"
            >
              Armenia
            </option>
            <option
              key="Aruba"
            >
              Aruba
            </option>
            <option
              key="Australia"
            >
              Australia
            </option>
            <option
              key="Austria"
            >
              Austria
            </option>
            <option
              key="Azerbaijan"
            >
              Azerbaijan
            </option>
            <option
              key="Bahamas"
            >
              Bahamas
            </option>
            <option
              key="Bahrain"
            >
              Bahrain
            </option>
            <option
              key="Bangladesh"
            >
              Bangladesh
            </option>
            <option
              key="Barbados"
            >
              Barbados
            </option>
            <option
              key="Belarus"
            >
              Belarus
            </option>
            <option
              key="Belgium"
            >
              Belgium
            </option>
            <option
              key="Belize"
            >
              Belize
            </option>
            <option
              key="Benin"
            >
              Benin
            </option>
            <option
              key="Bermuda"
            >
              Bermuda
            </option>
            <option
              key="Bhutan"
            >
              Bhutan
            </option>
            <option
              key="Bolivia, Plurinational State of"
            >
              Bolivia, Plurinational State of
            </option>
            <option
              key="Bonaire, Sint Eustatius and Saba"
            >
              Bonaire, Sint Eustatius and Saba
            </option>
            <option
              key="Bosnia and Herzegovina"
            >
              Bosnia and Herzegovina
            </option>
            <option
              key="Botswana"
            >
              Botswana
            </option>
            <option
              key="Bouvet Island"
            >
              Bouvet Island
            </option>
            <option
              key="Brazil"
            >
              Brazil
            </option>
            <option
              key="British Indian Ocean Territory"
            >
              British Indian Ocean Territory
            </option>
            <option
              key="Brunei Darussalam"
            >
              Brunei Darussalam
            </option>
            <option
              key="Bulgaria"
            >
              Bulgaria
            </option>
            <option
              key="Burkina Faso"
            >
              Burkina Faso
            </option>
            <option
              key="Burundi"
            >
              Burundi
            </option>
            <option
              key="Cabo Verde"
            >
              Cabo Verde
            </option>
            <option
              key="Cambodia"
            >
              Cambodia
            </option>
            <option
              key="Cameroon"
            >
              Cameroon
            </option>
            <option
              key="Canada"
            >
              Canada
            </option>
            <option
              key="Cayman Islands"
            >
              Cayman Islands
            </option>
            <option
              key="Central African Republic"
            >
              Central African Republic
            </option>
            <option
              key="Chad"
            >
              Chad
            </option>
            <option
              key="Chile"
            >
              Chile
            </option>
            <option
              key="China"
            >
              China
            </option>
            <option
              key="Christmas Island"
            >
              Christmas Island
            </option>
            <option
              key="Cocos (Keeling) Islands"
            >
              Cocos (Keeling) Islands
            </option>
            <option
              key="Colombia"
            >
              Colombia
            </option>
            <option
              key="Comoros"
            >
              Comoros
            </option>
            <option
              key="Congo"
            >
              Congo
            </option>
            <option
              key="Congo, Democratic Republic of the"
            >
              Congo, Democratic Republic of the
            </option>
            <option
              key="Cook Islands"
            >
              Cook Islands
            </option>
            <option
              key="Costa Rica"
            >
              Costa Rica
            </option>
            <option
              key="Croatia"
            >
              Croatia
            </option>
            <option
              key="Cuba"
            >
              Cuba
            </option>
            <option
              key="Curaçao"
            >
              Curaçao
            </option>
            <option
              key="Cyprus"
            >
              Cyprus
            </option>
            <option
              key="Czechia"
            >
              Czechia
            </option>
            <option
              key="Côte d'Ivoire"
            >
              Côte d'Ivoire
            </option>
            <option
              key="Denmark"
            >
              Denmark
            </option>
            <option
              key="Djibouti"
            >
              Djibouti
            </option>
            <option
              key="Dominica"
            >
              Dominica
            </option>
            <option
              key="Dominican Republic"
            >
              Dominican Republic
            </option>
            <option
              key="Ecuador"
            >
              Ecuador
            </option>
            <option
              key="Egypt"
            >
              Egypt
            </option>
            <option
              key="El Salvador"
            >
              El Salvador
            </option>
            <option
              key="Equatorial Guinea"
            >
              Equatorial Guinea
            </option>
            <option
              key="Eritrea"
            >
              Eritrea
            </option>
            <option
              key="Estonia"
            >
              Estonia
            </option>
            <option
              key="Eswatini"
            >
              Eswatini
            </option>
            <option
              key="Ethiopia"
            >
              Ethiopia
            </option>
            <option
              key="Falkland Islands (Malvinas)"
            >
              Falkland Islands (Malvinas)
            </option>
            <option
              key="Faroe Islands"
            >
              Faroe Islands
            </option>
            <option
              key="Fiji"
            >
              Fiji
            </option>
            <option
              key="Finland"
            >
              Finland
            </option>
            <option
              key="France"
            >
              France
            </option>
            <option
              key="French Guiana"
            >
              French Guiana
            </option>
            <option
              key="French Polynesia"
            >
              French Polynesia
            </option>
            <option
              key="French Southern Territories"
            >
              French Southern Territories
            </option>
            <option
              key="Gabon"
            >
              Gabon
            </option>
            <option
              key="Gambia"
            >
              Gambia
            </option>
            <option
              key="Georgia"
            >
              Georgia
            </option>
            <option
              key="Germany"
            >
              Germany
            </option>
            <option
              key="Ghana"
            >
              Ghana
            </option>
            <option
              key="Gibraltar"
            >
              Gibraltar
            </option>
            <option
              key="Greece"
            >
              Greece
            </option>
            <option
              key="Greenland"
            >
              Greenland
            </option>
            <option
              key="Grenada"
            >
              Grenada
            </option>
            <option
              key="Guadeloupe"
            >
              Guadeloupe
            </option>
            <option
              key="Guam"
            >
              Guam
            </option>
            <option
              key="Guatemala"
            >
              Guatemala
            </option>
            <option
              key="Guernsey"
            >
              Guernsey
            </option>
            <option
              key="Guinea"
            >
              Guinea
            </option>
            <option
              key="Guinea-Bissau"
            >
              Guinea-Bissau
            </option>
            <option
              key="Guyana"
            >
              Guyana
            </option>
            <option
              key="Haiti"
            >
              Haiti
            </option>
            <option
              key="Heard Island and McDonald Islands"
            >
              Heard Island and McDonald Islands
            </option>
            <option
              key="Holy See"
            >
              Holy See
            </option>
            <option
              key="Honduras"
            >
              Honduras
            </option>
            <option
              key="Hong Kong"
            >
              Hong Kong
            </option>
            <option
              key="Hungary"
            >
              Hungary
            </option>
            <option
              key="Iceland"
            >
              Iceland
            </option>
            <option
              key="India"
            >
              India
            </option>
            <option
              key="Indonesia"
            >
              Indonesia
            </option>
            <option
              key="Iran, Islamic Republic of"
            >
              Iran, Islamic Republic of
            </option>
            <option
              key="Iraq"
            >
              Iraq
            </option>
            <option
              key="Ireland"
            >
              Ireland
            </option>
            <option
              key="Isle of Man"
            >
              Isle of Man
            </option>
            <option
              key="Israel"
            >
              Israel
            </option>
            <option
              key="Italy"
            >
              Italy
            </option>
            <option
              key="Jamaica"
            >
              Jamaica
            </option>
            <option
              key="Japan"
            >
              Japan
            </option>
            <option
              key="Jersey"
            >
              Jersey
            </option>
            <option
              key="Jordan"
            >
              Jordan
            </option>
            <option
              key="Kazakhstan"
            >
              Kazakhstan
            </option>
            <option
              key="Kenya"
            >
              Kenya
            </option>
            <option
              key="Kiribati"
            >
              Kiribati
            </option>
            <option
              key="Korea, Democratic People's Republic of"
            >
              Korea, Democratic People's Republic of
            </option>
            <option
              key="Korea, Republic of"
            >
              Korea, Republic of
            </option>
            <option
              key="Kuwait"
            >
              Kuwait
            </option>
            <option
              key="Kyrgyzstan"
            >
              Kyrgyzstan
            </option>
            <option
              key="Lao People's Democratic Republic"
            >
              Lao People's Democratic Republic
            </option>
            <option
              key="Latvia"
            >
              Latvia
            </option>
            <option
              key="Lebanon"
            >
              Lebanon
            </option>
            <option
              key="Lesotho"
            >
              Lesotho
            </option>
            <option
              key="Liberia"
            >
              Liberia
            </option>
            <option
              key="Libya"
            >
              Libya
            </option>
            <option
              key="Liechtenstein"
            >
              Liechtenstein
            </option>
            <option
              key="Lithuania"
            >
              Lithuania
            </option>
            <option
              key="Luxembourg"
            >
              Luxembourg
            </option>
            <option
              key="Macao"
            >
              Macao
            </option>
            <option
              key="Madagascar"
            >
              Madagascar
            </option>
            <option
              key="Malawi"
            >
              Malawi
            </option>
            <option
              key="Malaysia"
            >
              Malaysia
            </option>
            <option
              key="Maldives"
            >
              Maldives
            </option>
            <option
              key="Mali"
            >
              Mali
            </option>
            <option
              key="Malta"
            >
              Malta
            </option>
            <option
              key="Marshall Islands"
            >
              Marshall Islands
            </option>
            <option
              key="Martinique"
            >
              Martinique
            </option>
            <option
              key="Mauritania"
            >
              Mauritania
            </option>
            <option
              key="Mauritius"
            >
              Mauritius
            </option>
            <option
              key="Mayotte"
            >
              Mayotte
            </option>
            <option
              key="Mexico"
            >
              Mexico
            </option>
            <option
              key="Micronesia, Federated States of"
            >
              Micronesia, Federated States of
            </option>
            <option
              key="Moldova, Republic of"
            >
              Moldova, Republic of
            </option>
            <option
              key="Monaco"
            >
              Monaco
            </option>
            <option
              key="Mongolia"
            >
              Mongolia
            </option>
            <option
              key="Montenegro"
            >
              Montenegro
            </option>
            <option
              key="Montserrat"
            >
              Montserrat
            </option>
            <option
              key="Morocco"
            >
              Morocco
            </option>
            <option
              key="Mozambique"
            >
              Mozambique
            </option>
            <option
              key="Myanmar"
            >
              Myanmar
            </option>
            <option
              key="Namibia"
            >
              Namibia
            </option>
            <option
              key="Nauru"
            >
              Nauru
            </option>
            <option
              key="Nepal"
            >
              Nepal
            </option>
            <option
              key="Netherlands"
            >
              Netherlands
            </option>
            <option
              key="New Caledonia"
            >
              New Caledonia
            </option>
            <option
              key="New Zealand"
            >
              New Zealand
            </option>
            <option
              key="Nicaragua"
            >
              Nicaragua
            </option>
            <option
              key="Niger"
            >
              Niger
            </option>
            <option
              key="Nigeria"
            >
              Nigeria
            </option>
            <option
              key="Niue"
            >
              Niue
            </option>
            <option
              key="Norfolk Island"
            >
              Norfolk Island
            </option>
            <option
              key="North Macedonia"
            >
              North Macedonia
            </option>
            <option
              key="Northern Mariana Islands"
            >
              Northern Mariana Islands
            </option>
            <option
              key="Norway"
            >
              Norway
            </option>
            <option
              key="Oman"
            >
              Oman
            </option>
            <option
              key="Pakistan"
            >
              Pakistan
            </option>
            <option
              key="Palau"
            >
              Palau
            </option>
            <option
              key="Palestine, State of"
            >
              Palestine, State of
            </option>
            <option
              key="Panama"
            >
              Panama
            </option>
            <option
              key="Papua New Guinea"
            >
              Papua New Guinea
            </option>
            <option
              key="Paraguay"
            >
              Paraguay
            </option>
            <option
              key="Peru"
            >
              Peru
            </option>
            <option
              key="Philippines"
            >
              Philippines
            </option>
            <option
              key="Pitcairn"
            >
              Pitcairn
            </option>
            <option
              key="Poland"
            >
              Poland
            </option>
            <option
              key="Portugal"
            >
              Portugal
            </option>
            <option
              key="Puerto Rico"
            >
              Puerto Rico
            </option>
            <option
              key="Qatar"
            >
              Qatar
            </option>
            <option
              key="Romania"
            >
              Romania
            </option>
            <option
              key="Russian Federation"
            >
              Russian Federation
            </option>
            <option
              key="Rwanda"
            >
              Rwanda
            </option>
            <option
              key="Réunion"
            >
              Réunion
            </option>
            <option
              key="Saint Barthélemy"
            >
              Saint Barthélemy
            </option>
            <option
              key="Saint Helena, Ascension and Tristan da Cunha"
            >
              Saint Helena, Ascension and Tristan da Cunha
            </option>
            <option
              key="Saint Kitts and Nevis"
            >
              Saint Kitts and Nevis
            </option>
            <option
              key="Saint Lucia"
            >
              Saint Lucia
            </option>
            <option
              key="Saint Martin (French part)"
            >
              Saint Martin (French part)
            </option>
            <option
              key="Saint Pierre and Miquelon"
            >
              Saint Pierre and Miquelon
            </option>
            <option
              key="Saint Vincent and the Grenadines"
            >
              Saint Vincent and the Grenadines
            </option>
            <option
              key="Samoa"
            >
              Samoa
            </option>
            <option
              key="San Marino"
            >
              San Marino
            </option>
            <option
              key="Sao Tome and Principe"
            >
              Sao Tome and Principe
            </option>
            <option
              key="Saudi Arabia"
            >
              Saudi Arabia
            </option>
            <option
              key="Senegal"
            >
              Senegal
            </option>
            <option
              key="Serbia"
            >
              Serbia
            </option>
            <option
              key="Seychelles"
            >
              Seychelles
            </option>
            <option
              key="Sierra Leone"
            >
              Sierra Leone
            </option>
            <option
              key="Singapore"
            >
              Singapore
            </option>
            <option
              key="Sint Maarten (Dutch part)"
            >
              Sint Maarten (Dutch part)
            </option>
            <option
              key="Slovakia"
            >
              Slovakia
            </option>
            <option
              key="Slovenia"
            >
              Slovenia
            </option>
            <option
              key="Solomon Islands"
            >
              Solomon Islands
            </option>
            <option
              key="Somalia"
            >
              Somalia
            </option>
            <option
              key="South Africa"
            >
              South Africa
            </option>
            <option
              key="South Georgia and the South Sandwich Islands"
            >
              South Georgia and the South Sandwich Islands
            </option>
            <option
              key="South Sudan"
            >
              South Sudan
            </option>
            <option
              key="Spain"
            >
              Spain
            </option>
            <option
              key="Sri Lanka"
            >
              Sri Lanka
            </option>
            <option
              key="Sudan"
            >
              Sudan
            </option>
            <option
              key="Suriname"
            >
              Suriname
            </option>
            <option
              key="Svalbard and Jan Mayen"
            >
              Svalbard and Jan Mayen
            </option>
            <option
              key="Sweden"
            >
              Sweden
            </option>
            <option
              key="Switzerland"
            >
              Switzerland
            </option>
            <option
              key="Syrian Arab Republic"
            >
              Syrian Arab Republic
            </option>
            <option
              key="Taiwan, Province of China"
            >
              Taiwan, Province of China
            </option>
            <option
              key="Tajikistan"
            >
              Tajikistan
            </option>
            <option
              key="Tanzania, United Republic of"
            >
              Tanzania, United Republic of
            </option>
            <option
              key="Thailand"
            >
              Thailand
            </option>
            <option
              key="Timor-Leste"
            >
              Timor-Leste
            </option>
            <option
              key="Togo"
            >
              Togo
            </option>
            <option
              key="Tokelau"
            >
              Tokelau
            </option>
            <option
              key="Tonga"
            >
              Tonga
            </option>
            <option
              key="Trinidad and Tobago"
            >
              Trinidad and Tobago
            </option>
            <option
              key="Tunisia"
            >
              Tunisia
            </option>
            <option
              key="Turkey"
            >
              Turkey
            </option>
            <option
              key="Turkmenistan"
            >
              Turkmenistan
            </option>
            <option
              key="Turks and Caicos Islands"
            >
              Turks and Caicos Islands
            </option>
            <option
              key="Tuvalu"
            >
              Tuvalu
            </option>
            <option
              key="Uganda"
            >
              Uganda
            </option>
            <option
              key="Ukraine"
            >
              Ukraine
            </option>
            <option
              key="United Arab Emirates"
            >
              United Arab Emirates
            </option>
            <option
              key="United Kingdom"
            >
              United Kingdom
            </option>
            <option
              key="United States Minor Outlying Islands"
            >
              United States Minor Outlying Islands
            </option>
            <option
              key="United States"
            >
              United States
            </option>
            <option
              key="Uruguay"
            >
              Uruguay
            </option>
            <option
              key="Uzbekistan"
            >
              Uzbekistan
            </option>
            <option
              key="Vanuatu"
            >
              Vanuatu
            </option>
            <option
              key="Venezuela, Bolivarian Republic of"
            >
              Venezuela, Bolivarian Republic of
            </option>
            <option
              key="Viet Nam"
            >
              Viet Nam
            </option>
            <option
              key="Virgin Islands, British"
            >
              Virgin Islands, British
            </option>
            <option
              key="Virgin Islands, U.S."
            >
              Virgin Islands, U.S.
            </option>
            <option
              key="Wallis and Futuna"
            >
              Wallis and Futuna
            </option>
            <option
              key="Western Sahara"
            >
              Western Sahara
            </option>
            <option
              key="Yemen"
            >
              Yemen
            </option>
            <option
              key="Zambia"
            >
              Zambia
            </option>
            <option
              key="Zimbabwe"
            >
              Zimbabwe
            </option>
          </FormControl>
          <Tooltip
            errors={Object {}}
            fieldName="country"
            forwardRef={
              Object {
                "current": null,
              }
            }
            touched={Object {}}
          />
        </FormGroup>
      </Col>
      <Col
        lg={5}
      >
        <FormGroup
          controlId="zip"
        >
          <FormControl
            name="zip"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="ZIP"
            type="text"
            value=""
          />
          <Tooltip
            errors={Object {}}
            fieldName="zip"
            forwardRef={
              Object {
                "current": null,
              }
            }
            touched={Object {}}
          />
        </FormGroup>
      </Col>
    </FormRow>
    <Styled(Button)
      className="px-4 px-sm-5 py-2"
      type="submit"
    >
      Continue
    </Styled(Button)>
  </Form>
</Fragment>
`;
